iterative/utils/helpers.go

```diff
@@ -7,6 +7,29 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+var (
+	SynthRegions = map[string]map[string]string{
```

Comment:
Not sure about this... I prefer to have the GetRegions implementation per provider. Or should go in the machine... if we were using the machine...

Reply:
got it

Reply:
I moved it here because I think there was actually a bug in the subnet-id code after you guys fixed the last one.

See previous:
```go
	region := GetRegion(d.Get("region").(string))
	availabilityZone := GetAvailabilityZone(region)
```
The get avail func would always get a stripped region and never detect an availability-zone, now it needs to first reference the synth-regions map to eliminate those as a false positive.

Since it need to be referenced in multiple places I went ahead and broke it out / consolidated that helper func across all providers.